core/store/migrate/migrations/0259_add_workflow_artifacts.sql

```diff
@@ -0,0 +1,43 @@
+-- +goose Up
+-- +goose StatementBegin
+-- Create the workflow_artifacts table
+CREATE TABLE workflow_artifacts (
+    id SERIAL PRIMARY KEY,
+    secrets_url TEXT,
+    secrets_url_hash TEXT UNIQUE,
+    contents BYTEA
+);
+
+-- Create an index on the secrets_url_hash column
+CREATE INDEX idx_secrets_url_hash ON workflow_artifacts(secrets_url_hash);
+
+-- Alter the workflow_specs table
+ALTER TABLE workflow_specs
+ADD COLUMN binary_url TEXT,
+ADD COLUMN artifact_id INT UNIQUE REFERENCES workflow_artifacts(id) ON DELETE CASCADE;
+
+-- Alter the config column type
+ALTER TABLE workflow_specs
+ALTER COLUMN config TYPE TEXT;
+
+-- Set the default value for the config column
+ALTER TABLE workflow_specs
+ALTER COLUMN config SET DEFAULT '';
```

Comment:
My PSQL knowledge is a bit rusty, but adding a default (as opposed to making leaving it nullable) will require a lock on the entire table + the modification of each row. Depending on how large the table is that could be more or less dangerous.

In this case I think it's fine since we're only running this on staging and the tables are small.